### PR TITLE
Simplify down to one stack.yaml

### PR DESCRIPTION
**Pull request by michaelpj**
As discussed on Matrix.

Since we don't use this for user installation, nor for testing multiple
versions of GHC (we have Haskell-CI based on `cabal` for that), the only
use of this is to provide a dev setup for stack users. As such, one
`stack.yaml` should be sufficient.

I checked and `stack build` still succeeds.

### CODE REVIEW

**Comment by michaelpj** [comment]
Bumping the install-nix-action version seems to fix the nix issue, the bump is in https://github.com/haskell/lsp/pull/378, I'll rebase this after taht's in.

**Review by jneira — APPROVED** [review]
nice! the remaining stack.yaml is almost perfect

File: stack.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+packages:
+- ./lsp
+- ./lsp-types
+- ./lsp-test
+
+flags: {}
+extra-package-dbs: []
+nix:
+  packages: [icu]
```
Comment (Ailrun) [inline stack.yaml:11]: Wow, how clean it is! Thanks!

**Review by Ailrun — APPROVED** [review]

**Comment by michaelpj** [comment]
It's only clean because 8.10 is so old now that lots of packages have got into stackage :P

**Comment by jneira** [comment]
...and LSP packages and it's deps live in stackage

**Pull request merged**